bin/templates/cordova/lib/build.js

```diff
@@ -422,11 +500,18 @@ function parseOpts(options, resolvedTarget) {
 module.exports.runClean = function(options) {
     var opts = parseOpts(options);
     var builder = builders[opts.buildMethod];
-    return builder.prepEnv()
+    return builder.prepEnv(opts)
     .then(function() {
-        return builder.clean(opts.extraArgs);
+        return builder.clean(opts);
     }).then(function() {
         shell.rm('-rf', path.join(ROOT, 'out'));
+
+        ['debug', 'release'].forEach(function(config) {
+            var propertiesFilePath = path.join(ROOT, config + SIGNING_PROPERTIES);
+            if(isAutoGenerated(propertiesFilePath)){
+                shell.rm('-f', propertiesFilePath);
```

Comment:
prepEnv (which is called in this case) takes care of deleting these if there is no supplied signing args. I don't think there's a need to delete them unconditionally since we don't bother deleting other build-time-supplied-files (like build.xml, build.gradle)